backend/server/app.py

```
@@ -87,57 +123,63 @@ async def get_latest_vol_surface(
+            interpolated_data, 
+            min_moneyness, 
+            max_moneyness, 
+            min_maturity, 
+            max_maturity
+        )
+        return JSONResponse(content=filtered)
     except Exception as e:
         raise HTTPException(
             status_code=500,
             detail=f"Error interpolating surface: {str(e)}"
         )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)